### PR TITLE
[Recipe] Serialization #9

### PR DESCRIPTION
**Pull request by FrankC01**

### CODE REVIEW

**Review by Jacob-CG — COMMENTED** [review]
Awesome work! Just a few comments

File: code/serialization/clientdata/ts.client.data.en.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+const borsh = require("borsh");
```
[the remark this answers is not in the document]
Comment (FrankC01) [reply code/serialization/clientdata/ts.client.data.en.ts:1]: I will change

File: docs/recipes/serialization.md [hunk 2]
```diff
@@ -1,3 +1,196 @@
...
+It is important that the above actions are all supported by the same serialization approach. The
+included snippets are demonstrating serializaation using [Borsh][1] with examples in Rust and Typescript.
+
+The samples in the remainder of this document are excerpts as taken from this [Solana Template][2]
+
+In this sample program, there are 4 operations, identified by their Instruction variant index:
+1. Initialize an account (variant index = 0)
+2. Mint a key value pair for a single account (variant index = 1)
+3. Transfer a key value pair from one account to another (variant index = 2)
+4. Burn a key value pair from an account (variant index = 3)
```
[the remark this answers is not in the document]
Comment (FrankC01) [reply docs/recipes/serialization.md:25]: OK, I can remove

File: docs/recipes/serialization.md [hunk 3]
```diff
@@ -1,3 +1,196 @@
...
+# Serialization
+
+When we talk about serialization we mean both serializing data as well as deserialization of data.
+
+Serialization comes into play at a few points along Solana program and program accounts lifecycle:
+
+1. Serializing instruction data (Client)
+2. Deserializing instruction data (Program)
+3. Account Data Serialization (Program)
+4. Client Account Data Deserialization (Client)
```
[the remark this answers is not in the document]
Comment (FrankC01) [reply docs/recipes/serialization.md:14]: Won't touch this

File: docs/recipes/serialization.md [hunk 4]
```diff
@@ -1,3 +1,196 @@
...
+
+It is important that the above actions are all supported by the same serialization approach. The
+included snippets are demonstrating serializaation using [Borsh][1] with examples in Rust and Typescript.
+
+The samples in the remainder of this document are excerpts as taken from this [Solana Template][2]
```
[the remark this answers is not in the document]
Comment (FrankC01) [reply docs/recipes/serialization.md:19]: Not sure what you mean. Do you mean create a Resource header at bottom of page and and just put the links there?

**Pull request merged**